### DISCUSSION TITLE
jupyterlab-quarto - previewing the quarto project on JupyterHub

### DISCUSSION THREAD

**Discussion by jgunstone** [quarto-dev/quarto-cli#10555] [Q&A]
Hi there, 

I :heart: quarto. 
I manage a JupyterHub and I'd like to set up a repo2docker that is preconfigured as a quarto project and show users how to use quarto to build a website.

This requires the use of `quarto preview .` in the docker container. 
Is this possible in JupyterHub? 
I'm struggling with it and it seems other uses have also had problems in the past: https://discourse.jupyter.org/t/previewing-quarto-rendering-from-jupyter-hub/15505

**Comment by cscheid** [comment]
(I transferred your question from the `quarto-dev/quarto` repo to here. In the future, please start these in our discussions forum here. Thank you!)

**Reply by jgunstone** [reply]
will do - thanks

**Comment by mcanouil** [comment]
> Is this possible in JupyterHub?
> I'm struggling with it and it seems other uses have also had problems in the past: https://discourse.jupyter.org/t/previewing-quarto-rendering-from-jupyter-hub/15505

I've never used JupyterHub, but since it works (as far as I know) in a Docker container classically  deployed with Docker with no additional layers, it's very likely something on JupyterHub.
You should check JupyterHub for anything regarding port forwarding, etc.

Also you might want to set some flags for `quarto preview` or options in `_quarto.yml`.
See <https://quarto.org/docs/reference/projects/options.html#preview>.
